pkg/stanza/adapter/converter.go

```diff
@@ -156,43 +156,48 @@ func (c *Converter) workerLoop() {
 
 	for entries := range c.workerChan {
 
-		resourceHashToIdx := make(map[uint64]int)
-		scopeIdxByResource := make(map[uint64]map[string]int)
+		pLogs := ConvertEntries(entries)
 
-		pLogs := plog.NewLogs()
-		var sl plog.ScopeLogs
+		// Send plogs directly to flushChan
+		c.flushChan <- pLogs
```

Comment:
nit: might as well save the allocation
```suggestion
		c.flushChan <- ConvertEntries(entries)
```